api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -147,8 +147,25 @@ default TableScan select(String... columns) {
    * @param toSnapshotId read append data up to this snapshot id
    * @return a table scan which can read append data from {@code fromSnapshotId}
    * exclusive and up to {@code toSnapshotId} inclusive
+   *
+   * @deprecated Please use the appendsBetween with non-primitive fromSnapshotId of Long type
+   */
+  @Deprecated
+  default TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+    return appendsInRange(Long.valueOf(fromSnapshotId), toSnapshotId);
+  }
+
+  /**
+   * Create a new {@link TableScan} to read appended data from {@code fromSnapshotId} exclusive to {@code toSnapshotId}
+   * inclusive.
+   *
+   * @param fromSnapshotId the last snapshot id read by the user, exclusive. If it is null,
+   *                       all the ancestor snapshots of the toSnapshotId should be discovered
```

Comment:
do you mean all ancestor snapshots should be included? right?

Reply:
correct. let's say the snapshot history is Sn -> Sn+1 -> Sn+2 -> Sn+3. `appendsBetween(null, Sn+3) should discover files appended by all 4 snapshots.